src/hooks/useObserverToGatewayMap.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import useGateways from './useGateways';
+
+const useObserverToGatewayMap = () => {
+  const { data: gateways } = useGateways();
+  const [observerToGatewayMap, setObserverToGatewayMap] =
+    useState<Record<string, string>>();
+
+  useEffect(() => {
+    if (gateways) {
+      const results: Record<string, string> = {};
+      Object.entries(gateways).forEach(([gatewayAddress, gateway]) => {
+        results[gateway.observerAddress] = gatewayAddress;
+      });
+      setObserverToGatewayMap(results);
+    }
+  }, [gateways]);
+
+  return observerToGatewayMap;
```

Comment:
should be able to do the following without a useEffect. Less hooks good.
```suggestion
return Object.entries(gateways).reduce((acc, [gatewayAddress, gateway]) => {
  acc[gateway.observerAddress] = gatewayAddress;
  return acc;
}, {} as Record<string, string>);
```

Reply:
Using the useEffect() reduces repeated work here.